src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +501,128 @@ public override async Task RunAsync()
+                        ColoredConsole.WriteLine(VerboseColor($"Selected {DotnetConstants.InProc6HostRuntime} host"));
+                    }
+                }
+                return true;
+            }
+            // If the above conditions fail, the default should be OOP host
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor("Selected out-of-process host"));
+            }
+            return false;
+          
+        }
+
         private static string GetInProcNet8ExecutablePath()
         {
             var funcExecutableDirectory = Path.GetDirectoryName(typeof(StartHostAction).Assembly.Location)!;
-            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? WindowsExecutableName : LinuxExecutableName;
+            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DotnetConstants.WindowsExecutableName : DotnetConstants.LinuxExecutableName;
+
+            return Path.Combine(funcExecutableDirectory, DotnetConstants.InProc8DirectoryName, executableName);
+        }
+
+        private static string GetInProcNet6ExecutablePath()
+        {
+            var funcExecutableDirectory = Path.GetDirectoryName(typeof(StartHostAction).Assembly.Location)!;
+            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DotnetConstants.WindowsExecutableName : DotnetConstants.LinuxExecutableName;
 
-            return Path.Combine(funcExecutableDirectory, InProc8DirectoryName, executableName);
+            return Path.Combine(funcExecutableDirectory, DotnetConstants.InProc6DirectoryName, executableName);
         }
 
-        private Task StartInProc8AsChildProcessAsync()
+        private async Task StartHostAsChildProcessAsync(bool shouldStartNet8ChildProcess)
         {
             if (VerboseLogging == true)
             {
-                ColoredConsole.WriteLine(VerboseColor($"Starting child process for in-process model host."));
+                ColoredConsole.WriteLine(VerboseColor($"Starting child process for {(shouldStartNet8ChildProcess ? DotnetConstants.InProc8HostRuntime : DotnetConstants.InProc6HostRuntime)} model host."));
             }
 
             var commandLineArguments = string.Join(" ", Environment.GetCommandLineArgs().Skip(1));
             var tcs = new TaskCompletionSource();
 
-            var inProc8FuncExecutablePath = GetInProcNet8ExecutablePath();
+            var funcExecutablePath = shouldStartNet8ChildProcess ? GetInProcNet8ExecutablePath() : GetInProcNet6ExecutablePath();
 
-            EnsureNet8FuncExecutablePresent(inProc8FuncExecutablePath);
+            EnsureFuncExecutablePresent(funcExecutablePath, shouldStartNet8ChildProcess);
 
-            var inprocNet8ChildProcessInfo = new ProcessStartInfo
+            var childProcessInfo = new ProcessStartInfo
             {
-                FileName = inProc8FuncExecutablePath,
+                FileName = funcExecutablePath,
```

Comment:
Where do we use the `--no-build` argument?

Reply:
We use it in line 626?